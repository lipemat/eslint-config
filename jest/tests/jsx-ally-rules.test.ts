import jestRunnerEslint from '../../jest/helpers/jest-runner-eslint';

describe( 'JSX A11Y rules are enabled', () => {
	test( 'jsx-a11y/label-has-associated-control', async () => {
		expect( await jestRunnerEslint( 'passing/jsx-ally/label-has-associated-control' ) ).toMatchSnapshot();

		expect( await jestRunnerEslint( 'failing/jsx-ally/label-has-associated-control' ) ).toMatchSnapshot();
	} );
} );
