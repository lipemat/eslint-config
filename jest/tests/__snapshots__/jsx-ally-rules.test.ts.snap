// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX A11Y rules are enabled jsx-a11y/label-has-associated-control 1`] = `
"PASS jest/fixtures/passing/jsx-ally/label-has-associated-control/custom-component.tsx
  √ ESLint (mocked ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        
Ran all test suites.
"
`;

exports[`JSX A11Y rules are enabled jsx-a11y/label-has-associated-control 2`] = `
"FAIL jest/fixtures/failing/jsx-ally/label-has-associated-control/not-associated.tsx
  × jsx-a11y/label-has-associated-control (mocked ms)


E:\\SVN\\the-boilerplate\\packages\\eslint-config\\jest\\fixtures\\failing\\jsx-ally\\label-has-associated-control\\not-associated.tsx
  4:4  error  A form label must be associated with a control  jsx-a11y/label-has-associated-control

✖ 1 problem (1 error, 0 warnings)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        
Ran all test suites.
"
`;
